estimator/util.py

```diff
@@ -181,6 +181,51 @@ def binary_search(
         return it.y
 
 
+def robust_bin_search(
```

Comment:
Shall we absorb this into `local_minimum`?

Reply:
I found `local_minimum` already somewhat complicated to understand, so including another layer of complexity might be bad for readability. We could replace `binary_search` by this though, since the default setting `step=1` is essentially the same thing.

Reply:
I personally find `with local_minium() as it: blah` much easier to read so I might try to turn your function into an iterator and then have `with local_minimum_simple` and `with local_minimum`?

Reply:
I agree that `with local_minimum as it` is nice and easy to read, I just find the implementation of the class `local_minimum` not easy to follow and worry that including the step in there might make it even trickier. Especially, since it is very easy to implement this feature as a wrapper (as is done in `robust_bin_search`) and this provides a nice layer of abstraction. But maybe writing a new context class that uses the current one might be even more elegant.